geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaIndex.java

```diff
@@ -0,0 +1,35 @@
+package org.locationtech.geomesa.api;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import java.util.Date;
+
+/**
+ * GeoMesaIndex is an API for utilizing GeoMesa as a spatial index
+ * without bringing in the Geotools SimpleFeature data model
+ * @param <T>
+ */
+public interface GeoMesaIndex<T> {
+
+    /**
+     * Query a GeoMesa index
+     * @param query
+     * @return
+     */
+    Iterable<T> query(GeoMesaQuery query);
+
+    /**
+     * Put a value in the GeoMesa index
+     * @param t
+     * @param geometry
+     * @param dtg
+     */
+    void put(T t, Geometry geometry, Date dtg);
+
+    /**
+     * Delete a value from the GeoMesa index
+     * @param t
+     */
+    void delete(T t);
```

Comment:
delete gets easier if you go with the feature ID :) since geomesa can support that. However, if you only have the Z3 index the delete of a named object is harder and the better syntax would be something like ```delete where <query>``` like you do in SQL. Else you'd have to keep a ref to the object identifier somehow since you'd have no way to find it in the Z3 index alone and we certainly don't want to store a records table in this case I don't think.